In [1]:
#imports
import pandas as pd
import numpy as np

# suppress scientific notation by setting float_format
pd.options.display.float_format = '{:.0f}'.format

# Setting the warnings to be ignored 
import warnings 
warnings.filterwarnings('ignore')

In [2]:
# loading in data for year 2000
df2000 = pd.read_csv("Data/final_tmdb_data_2000.csv.gz")
df2000.head()

imdb_id  adult                     backdrop_path belongs_to_collection  \
0          0    NaN                               NaN                   NaN   
1  tt0113026      0  /vMFs7nw6P0bIV1jDsQpxAieAVnH.jpg                   NaN   
2  tt0113092      0                               NaN                   NaN   
3  tt0116391      0                               NaN                   NaN   
4  tt0116748      0  /wr0hTHwkYIRC82MwNbhOvqrw27N.jpg                   NaN   

    budget                                             genres homepage     id  \
0      NaN                                                NaN      NaN    NaN   
1 10000000  [{'id': 35, 'name': 'Comedy'}, {'id': 10402, '...      NaN  62127   
2        0           [{'id': 878, 'name': 'Science Fiction'}]      NaN 110977   
3        0  [{'id': 18, 'name': 'Drama'}, {'id': 28, 'name...      NaN 442869   
4        0  [{'id': 18, 'name': 'Drama'}, {'id': 10749, 'n...      NaN 579396   

  original_language   original_title  ... revenue  runtime  \
0               NaN              NaN  ...     NaN      NaN   
1                en  The Fantasticks  ...       0       86   
2                en    For the Cause  ...       0      100   
3                hi             Gang  ...       0      152   
4                hi         Karobaar  ...       0      180   

                                    spoken_languages    status  \
0                                                NaN       NaN   
1  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
2  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
3  [{'english_name': 'Hindi', 'iso_639_1': 'hi', ...  Released   
4  [{'english_name': 'Hindi', 'iso_639_1': 'hi', ...  Released   

                                         tagline            title  video  \
0                                            NaN              NaN    NaN   
1  Try to remember the first time magic happened  The Fantasticks      0   
2   The ultimate showdown on a forbidden planet.    For the Cause      0   
3                                            NaN             Gang      0   
4                           The Business of Love         Karobaar      0   

   vote_average vote_count certification  
0           NaN        NaN           NaN  
1             6         22           NaN  
2             5         10           NaN  
3             4          1           NaN  
4             7          3           NaN  

[5 rows x 26 columns]

In [3]:
df2000.shape

(2253, 26)

In [4]:
# loading in data for year 2001
df2001 = pd.read_csv("Data/final_tmdb_data_2001.csv.gz")
df2001.head()

imdb_id  adult                     backdrop_path belongs_to_collection  \
0          0    NaN                               NaN                   NaN   
1  tt0035423      0  /tJLV3BAlHOgscVOrA99Wnb2gAef.jpg                   NaN   
2  tt0114447      0                               NaN                   NaN   
3  tt0114722      0                               NaN                   NaN   
4  tt0116916      0  /rFpHBidSlhjflmnLu7BZilyKeQR.jpg                   NaN   

    budget                                             genres homepage     id  \
0      NaN                                                NaN      NaN    NaN   
1 48000000  [{'id': 10749, 'name': 'Romance'}, {'id': 14, ...      NaN  11232   
2        0  [{'id': 53, 'name': 'Thriller'}, {'id': 28, 'n...      NaN 151007   
3        0  [{'id': 18, 'name': 'Drama'}, {'id': 53, 'name...      NaN 276251   
4        0  [{'id': 28, 'name': 'Action'}, {'id': 12, 'nam...      NaN  73549   

  original_language    original_title  ...  revenue  runtime  \
0               NaN               NaN  ...      NaN      NaN   
1                en    Kate & Leopold  ... 76019048      118   
2                en  The Silent Force  ...        0       90   
3                es          3 Noches  ...        0      105   
4                en     The Dark Mist  ...        0      101   

                                    spoken_languages    status  \
0                                                NaN       NaN   
1  [{'english_name': 'French', 'iso_639_1': 'fr',...  Released   
2  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
3  [{'english_name': 'Spanish', 'iso_639_1': 'es'...  Released   
4  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   

                                             tagline             title  video  \
0                                                NaN               NaN    NaN   
1  If they lived in the same century, they'd be p...    Kate & Leopold      0   
2  They left him for dead... They should have fin...  The Silent Force      0   
3                                                NaN          3 Nights      0   
4                                                NaN     The Dark Mist      0   

   vote_average vote_count certification  
0           NaN        NaN           NaN  
1             6       1236         PG-13  
2             5          3           NaN  
3             0          0           NaN  
4             4          2            PG  

[5 rows x 26 columns]

In [5]:
df2001.shape

(2335, 26)

In [6]:
# concatenating dataframes
dfcombined = pd.concat([df2000,df2001], axis = 0)
dfcombined.head()

imdb_id  adult                     backdrop_path belongs_to_collection  \
0          0    NaN                               NaN                   NaN   
1  tt0113026      0  /vMFs7nw6P0bIV1jDsQpxAieAVnH.jpg                   NaN   
2  tt0113092      0                               NaN                   NaN   
3  tt0116391      0                               NaN                   NaN   
4  tt0116748      0  /wr0hTHwkYIRC82MwNbhOvqrw27N.jpg                   NaN   

    budget                                             genres homepage     id  \
0      NaN                                                NaN      NaN    NaN   
1 10000000  [{'id': 35, 'name': 'Comedy'}, {'id': 10402, '...      NaN  62127   
2        0           [{'id': 878, 'name': 'Science Fiction'}]      NaN 110977   
3        0  [{'id': 18, 'name': 'Drama'}, {'id': 28, 'name...      NaN 442869   
4        0  [{'id': 18, 'name': 'Drama'}, {'id': 10749, 'n...      NaN 579396   

  original_language   original_title  ... revenue  runtime  \
0               NaN              NaN  ...     NaN      NaN   
1                en  The Fantasticks  ...       0       86   
2                en    For the Cause  ...       0      100   
3                hi             Gang  ...       0      152   
4                hi         Karobaar  ...       0      180   

                                    spoken_languages    status  \
0                                                NaN       NaN   
1  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
2  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
3  [{'english_name': 'Hindi', 'iso_639_1': 'hi', ...  Released   
4  [{'english_name': 'Hindi', 'iso_639_1': 'hi', ...  Released   

                                         tagline            title  video  \
0                                            NaN              NaN    NaN   
1  Try to remember the first time magic happened  The Fantasticks      0   
2   The ultimate showdown on a forbidden planet.    For the Cause      0   
3                                            NaN             Gang      0   
4                           The Business of Love         Karobaar      0   

   vote_average vote_count certification  
0           NaN        NaN           NaN  
1             6         22           NaN  
2             5         10           NaN  
3             4          1           NaN  
4             7          3           NaN  

[5 rows x 26 columns]

In [7]:
dfcombined.shape

(4588, 26)

How many movies had at least some valid financial information (values > 0 for budget OR revenue)?

In [8]:
dfvalidfininfo = dfcombined.loc[(dfcombined["revenue"] > 0.0) | (dfcombined["budget"] > 0.0)]
dfvalidfininfo.head()

imdb_id  adult                     backdrop_path  \
1   tt0113026      0  /vMFs7nw6P0bIV1jDsQpxAieAVnH.jpg   
5   tt0118694      0  /n4GJFGzsc7NinI1VeGDXIcQjtU2.jpg   
7   tt0119273      0  /f5C03doOWiauu37bToKXtpgP5bS.jpg   
13  tt0120202      0  /tuerGhZ1lA8wpBjUCEUpeTWDT5B.jpg   
14  tt0120263      0  /ynXVuylP8upazjz8lrqb1PEMkdR.jpg   

                                belongs_to_collection   budget  \
1                                                 NaN 10000000   
5                                                 NaN   150000   
7   {'id': 141086, 'name': 'Heavy Metal Collection... 15000000   
13                                                NaN        0   
14                                                NaN        0   

                                               genres  \
1   [{'id': 35, 'name': 'Comedy'}, {'id': 10402, '...   
5   [{'id': 18, 'name': 'Drama'}, {'id': 10749, 'n...   
7   [{'id': 28, 'name': 'Action'}, {'id': 12, 'nam...   
13  [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...   
14  [{'id': 18, 'name': 'Drama'}, {'id': 35, 'name...   

                                             homepage    id original_language  \
1                                                 NaN 62127                en   
5                                                 NaN   843                cn   
7                                                 NaN 16225                en   
13  http://www.newline.com/properties/stateandmain... 21991                en   
14                                                NaN 34070                sv   

                original_title  ...  revenue  runtime  \
1              The Fantasticks  ...        0       86   
5                         花樣年華  ... 14204632       99   
7             Heavy Metal 2000  ...        0       88   
13              State and Main  ...  9206279      106   
14  Sånger från andra våningen  ...    80334       99   

                                     spoken_languages    status  \
1   [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
5   [{'english_name': 'Cantonese', 'iso_639_1': 'c...  Released   
7   [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
13  [{'english_name': 'Italian', 'iso_639_1': 'it'...  Released   
14  [{'english_name': 'Russian', 'iso_639_1': 'ru'...  Released   

                                              tagline  \
1       Try to remember the first time magic happened   
5   Feel the heat, keep the feeling burning, let t...   
7   not to survive the fight in the external world...   
13               Big movie. Small town. Huge trouble.   
14                                                NaN   

                          title  video  vote_average vote_count certification  
1               The Fantasticks      0             6         22           NaN  
5          In the Mood for Love      0             8       2364            PG  
7              Heavy Metal 2000      0             6        193             R  
13               State and Main      0             6        184           NaN  
14  Songs from the Second Floor      0             7        249           NaN  

[5 rows x 26 columns]

In [9]:
dfvalidfininfo["imdb_id"].value_counts().sum()

756

There were 756 that had some valid financial informtation.

How many movies are there in each of the certification categories (G/PG/PG-13/R)?

In [10]:
dfvalidfininfo["certification"].value_counts()

R        230
PG-13    134
PG        36
NR        21
G         15
NC-17      1
Name: certification, dtype: int64

In [15]:
dfvalidfininfo["certification"].isna().sum().sum()

319

There are 319 missing values for the certification column.

What is the average revenue per certification category?

In [27]:
tempdf = dfvalidfininfo.groupby(["certification"]).mean()
tempdf[["revenue"]]

revenue
certification          
G             117364761
NC-17             11678
NR             16831559
PG            107729664
PG-13          97806800
R              30219318

What is the average budget per certification category?

In [26]:
tempdf[["budget"]]

budget
certification         
G             38133333
NC-17                0
NR             9184624
PG            44027701
PG-13         43004910
R             18440454

Saving final dataframe as a CSV file

In [18]:
dfvalidfininfo.to_csv('tmdb_results_combined.csv.gz', index=False, compression='gzip')